app.js

```diff
@@ -185,7 +186,48 @@ function MonitorApp ({
       });
     };
     
+    // Store in memcache db
+    // TODO: deprecate memcache db. just use mongo for everything.
     mjs.set(key, JSON.stringify(newRoutes), {}, handleErr);
+    
+    // Add new routes to mongo db log. Used for generating timeseries.
+    db.collection('routeLog').insertOne({
+      'timestamp': now,
+      // can omit timestamp from each route object since they're all the same
+      'routes': newRoutes.map((r) => _.omit(r, 'timestamp'))
+    });
+
+  }));
+
+  app.get('/api/v0/numNodesTimeseries', asyncMiddleware(async function(req, res) {
+    let numNodes = [];
+    let numGateways = [];
+    let timestamps = [];
+    
+    let now = new Date();
+    let yesterday = new Date(now - 1000 * 60 * 60 * 24);
+    let toDate = now;
+    let fromDate = yesterday;
+    if (req.query.from)
+      fromDate = new Date(req.query.from);
+    if (req.query.to)
+      toDate = new Date(req.query.to);
+
+    await db.collection('routeLog')
+      .find({
+        timestamp: {
+          '$lt': toDate,
+          '$gte': fromDate
+        }
+      })
+      .sort({ timestamp: 1 })
+      .forEach((routeLog) => {
+        timestamps.push(routeLog.timestamp);
+        numGateways.push(_.unique(routeLog.routes, (route) => route.gatewayIP).length);
+        numNodes.push(_.unique(routeLog.routes, (route) => route.nodeIP).length);
+      });
+
+    res.json({ numNodes, numGateways, timestamps });
```

Comment:
Wait so is `numGateways` expected to be an array of array of... numbers? Am I reading it right with the `push(_.unique(...))` above?

Would you, like me, normally expect `numX` to have a value that is a Number?

Reply:
fair enough :-P